src/liquid/components/ld-pagination/ld-pagination.tsx

```diff
@@ -0,0 +1,372 @@
+ * @part sticky - all sticky items (`ld-button` elements)
+ * @part wrapper - list containing all pagination items
+ */
+@Component({
+  assetsDirs: ['assets'],
+  tag: 'ld-pagination',
+  styleUrl: 'ld-pagination.css',
+  shadow: true,
+})
+export class LdPagination {
+  @Element() element: HTMLElement
+
+  /** Label to communicate the type of an item. */
+  @Prop() itemLabel = 'Page'
+
+  /** The maximum number of items. */
+  @Prop({ mutable: true }) length = Infinity
+
+  /** The currently selected item (an index of `-1` means nothing is selected). */
+  @Prop({ mutable: true }) selectedIndex = 0
+
+  /** Number of next/previous items visible. */
+  @Prop() offset = 2
+
+  /** Size of the pagination. */
+  @Prop() size?: 'sm' | 'lg'
+
+  /** Number of items permanently visible at the start/end. */
+  @Prop() sticky = 0
+
+  @State() maxSliderColumns = 0
+  @State() renderDots = false
+  @State() renderSticky = false
+  @State() slidableItems: number[] = []
+  @State() sliderContent: number[] = []
+  @State() visibleItemsInSlider = 0
+
+  /** Dispatched, if the selected index changes. */
+  @Event() ldchange: EventEmitter<number>
+
+  @Watch('selectedIndex')
+  handleSelectedIndex() {
+    if (this.selectedIndex < 0) {
+      this.selectedIndex = 0
+    } else if (this.selectedIndex >= this.length) {
+      this.selectedIndex = this.length - 1
+    } else {
+      this.ldchange.emit(this.selectedIndex)
+    }
+  }
+
+  // pageNumber is 1-based
+  private renderItem = (
+    itemNumber: number,
+    showFrom: number,
+    showTo: number
+  ) => {
+    const isHidden =
+      this.renderDots && (itemNumber < showFrom || itemNumber > showTo)
+    const isSelected = itemNumber === this.selectedIndex + 1
+    return (
+      <li
+        aria-hidden={isHidden ? 'true' : undefined}
+        class={getClassNames([
+          'ld-pagination__item',
+          !isHidden && 'ld-pagination__item--visible',
+          isSelected && 'ld-pagination__item--selected',
+        ])}
+        key={itemNumber}
+        style={{ '--ld-pagination-item-pos': `${itemNumber - 1}` }}
+      >
+        <ld-button
+          aria-current={isSelected ? 'true' : undefined}
+          aria-label={isHidden ? undefined : `${this.itemLabel} ${itemNumber}`}
+          ld-tabindex={isHidden ? -1 : undefined}
+          mode="ghost"
+          onClick={() => {
+            this.selectedIndex = itemNumber - 1
+          }}
+          part="item focusable"
+          size={this.size}
+        >
+          {itemNumber}
+        </ld-button>
+      </li>
+    )
+  }
+
+  @Watch('selectedIndex')
+  calculateSliderContent() {
+    const newSliderContent = this.slidableItems.filter(
+      (itemNumber) =>
+        itemNumber > this.selectedIndex - 20 &&
+        itemNumber < this.selectedIndex + 20
+    )
+
+    // Required because the rendering does not work properly and thus we
+    // have to await the animation before updating the slider content.
+    setTimeout(() => {
```

Comment:
Why not use [ontransitionend](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/ontransitionend)?
If you want to stick with `setTimeout`, you should also use `clearTimeout` in the `disconnectedCallback` hook.